### ISSUE TITLE
Monaco editor has no margin

### ISSUE THREAD

**Issue by kkga** [notable/notable#551]
- **OS Version**: Pop!_OS 10.18
- **Notable Version**: 1.4.0

### Current and expected behavior

Looks like switching to Monaco in the latest version removed the margin from the editor body. It would be nice to have back the consistent margin for editor and preview mode. 

![Peek 2019-03-15 14-42](https://user-images.githubusercontent.com/1460122/54429112-a8f2d680-4730-11e9-8c9e-b4374581f76f.gif)

**Comment by fabiospampinato** [comment]
Yeah that's something I miss too, unfortunately Monaco is great at many things but it doesn't excel in customizability, at all.

I've tried to fix this already actually, but I don't think it's fully fixable with the current state of the editor:

- In order to add some padding to the left I added an empty decorations gutter. The problem is that if you disable line wrapping and scroll the editor to the right that empty gutter remains there, fixed, creating kind of an awkward border.

- In order to add some padding to the right I told the editor that the scrollbar was like 20px wide and then resized it manually to 10px via CSS. This doesn't work when there's no scrollbar to be displayed, i.e. the document is short enough, it does also not work with line wrapping disabled, and although I haven't checked it I think it might mess up the position of the minimap too.

- In order to add some padding to the top one has to add a custom "view zone". This solution doesn't have any critical drawbacks. In fact I'm already using this for adding some padding to the top in zen mode in macOS, but even then it was pretty awkward to implement as things like "scroll to the top" didn't really work with it.

**Comment by fabiospampinato** [comment]
I've opened an issue in their repository for this: https://github.com/Microsoft/monaco-editor/issues/1368

**Comment by fabiospampinato** [comment]
I've kind of addressed this for the top and left margins, which should be enough for the majority of the cases. Adding margin to the right and bottom is either impossible or too buggy.

Here there are some screenshots, preview:

<img width="930" alt="Screen Shot 2019-05-17 at 02 14 15" src="https://user-images.githubusercontent.com/1812093/57895243-f384f200-784a-11e9-8bbe-14b76edd98a5.png">

Editor:

<img width="930" alt="Screen Shot 2019-05-17 at 02 14 20" src="https://user-images.githubusercontent.com/1812093/57895251-f8e23c80-784a-11e9-83b8-370a50352da0.png">

Editor + line numbers enabled:

<img width="930" alt="Screen Shot 2019-05-17 at 02 14 37" src="https://user-images.githubusercontent.com/1812093/57895252-f97ad300-784a-11e9-9141-4b19e1fdfe3f.png">

I'd be glad to hear your feedback on this folks, does it look better? Does it look like there's an empty line at the beginning and the whole thing is indented by default? 🤔

**Comment by jamiekyle-eb** [comment]
I really prefer the padding. I don't get the impression that there's an empty line or indentation.

I also really like the line numbers, maybe that would make a good setting people can turn on or off.

**Comment by fabiospampinato** [comment]
> I also really like the line numbers, maybe that would make a good setting people can turn on or off.

Eventually it will, once we have proper settings #476, for now I've added a shortcut for toggling it.
